components/Register.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { useRouter } from 'next/navigation';
+
+interface IFormInput {
+  email: string;
+  username: string;
+  password: string;
+  repeatPassword: string;
+}
+
+const RegisterForm: React.FC = () => {
+  const { register, handleSubmit, formState: { errors }, watch } = useForm<IFormInput>();
+
+  const router = useRouter();
+
+  const onSubmit: SubmitHandler<IFormInput> = async (data) => {
+    if (data.password !== data.repeatPassword) {
+      return alert('Passwords do not match');
+    }
+
+    const res = await fetch('/api/auth/register', {
+      method: 'POST',
+      body: JSON.stringify({
+        email: data.email,
+        username: data.username,
+        password: data.password
+      }),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+    //const resJSON = await res.json()
+    
+    if (res.ok) {
+      router.push('/auth/login')
+    }
+  };
+
+  const password = watch('password');
```

Comment:
Is there a reason for this line? Please explain, I am curious to know

Reply:
It is a function of react-hook-form to observe the password in real time and be able to check that it matches the password repetition. Even so, it is not doing it in real time, there must be something missing in the validation.